enhancements/installer/azurestack.md

```diff
@@ -0,0 +1,160 @@
+
+## Proposal
+
+
+### User Stories
+
+#### Story 1
+As a cluster admin, I want to be able to use `openshift-install` to create and destroy a cluster on
+Azure Stack Hub.
+
+#### Story 2
+As an OpenShift developer, I want Azure Stack Hub to be managed and released like other platforms and covered by CI tests.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Primary ways in which Azure Stack differs from Azure:
+- requires a different rhcos image
+- limited instance metadata service (IMDS) for VMs
+- does not support private DNS zones
+- limited subset of Azure infrastructure available (ergo different Terraform provider)
+
+### Risks and Mitigations
+
+Adding ASH support is somewhere in between adding a new Azure environment and an entirely new platform.
+
+## Design Details
+
+### Infrastructure Global Configuration
+
+#### API
+ASH will need to be added to the known [Azure Cloud Environments in the API](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L339-L355):
+
+```go
+// AzureStackCloud is the Azure cloud environment used on-premises.
+AzureStackCloud AzureCloudEnvironment = "AzureStackCloud"
+```
+
+#### Cluster Config Operator
+Cluster config operator will [need to be updated](https://github.com/openshift/cluster-config-operator/blob/master/pkg/operator/kube_cloud_config/azure.go#L23) and the new API vendored in.
+
+#### API endpoints
+ASH endpoints are user-provided and, therefore, the Azure SDK treats ASH endpoints differently than the already-known endpoints of other Azure environments. When the cloud environment is set to
+`AZURESTACKCLOUD` the SDK expects the environment variable `AZURE_ENVIRONMENT_FILEPATH` to point to a [json configuration file](https://kubernetes-sigs.github.io/cloud-provider-azure/install/configs/#azure-stack-configuration), which is [typically located at `/etc/kuberentes/azurestackcloud.json`](https://github.com/kubernetes-sigs/cloud-provider-azure/issues/151).
+
+The Installer will need to construct this json configuration file from user input and include the file as part of the cloud-provider-config configmap. The file will also need to be present on nodes for the kubelet. `AZURE_ENVIRONMENT_FILEPATH` will need to be set programmatically for ASH on the kubelet, presumably through the kubelet's systemd unit [(see open questions)](#open-questions).
+
+Operators will set the `AZURE_ENVIRONMENT_FILEPATH` and mount the endpoints JSON file from the cloud-provider-config configmap. All operators using the Azure SDK will need to do this.
+
+#### Installer Manifests & Cloud Provider Config
+ASH should be able to reuse Azure machine manifests with proper validation.
+
+The [Azure cloud-provider-config](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/azure/cloudproviderconfig.go) will need to be slightly adapted. `LoadBalancerSku` for ASH will be `basic`. `UseInstanceMetadata` should probably be `false`.
```

Comment:
`useManagedIdentityExtension` must be `false`. Need to explore further whether we an alternative method for auth is needed.